include/seqan3/range/container/small_string.hpp

```diff
@@ -0,0 +1,265 @@
+        assert(_lit[N - 1] == '\0');
+        clear();
+        assign(&_lit[0], &_lit[N-1]);
+    }
+    //!\}
+
+    /*!\name Capacity
+     * \{
+     */
+    //!\brief Returns the maximal size which equals the capacity.
+    constexpr size_type max_size() const noexcept
+    {
+        return capacity_;
+    }
+
+    //!\brief Returns the maximal capacity.
+    constexpr size_type capacity() const noexcept
+    {
+        return capacity_;
+    }
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Returns the content represented as std::string.
+     *
+     * \returns `std::string` The stored string.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee. No data is modified.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the string.
+     */
+    std::string string() const
+    {
+        return std::string{this->cbegin(), this->cend()};
+    }
+
+    /*!\brief Returns the content represented as 0-terminated c-style string.
+     *
+     * \returns `const char *` The stored string.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     */
+    constexpr const char * c_str() const noexcept
+    {
+        return this->data_.data();
+    }
+    //!\}
+
+    /*!\name Modifiers
+     * \{
+     */
+    //!\copydoc seqan3::small_vector::clear
+    constexpr void clear() noexcept
+    {
+        *this = small_string{};
+    }
+
+    //!\copydoc seqan3::small_vector::pop_back
+    constexpr void pop_back() noexcept
+    {
+        assert(this->sz > 0);
+        --this->sz;
+        this->data_[this->sz] = '\0';
+    }
+
+    //!\copydoc seqan3::small_vector::resize(size_type const)
+    constexpr void resize(size_type const count) noexcept
+    {
+        assert(count <= capacity_);
+
+        for (size_t i = this->sz; i <= count; ++i)
+            this->data_[i] = '\0';
+
+        this->sz = count;
+        this->data_[this->sz] = '\0';
+    }
+    //!\}
+
+
+    /*!\name Concatenation
+     * \{
+     */
+    /*!\brief Concatenates two small_strings by returning a new small_string.
+     * \param rhs The right-hand-side to concat with.
+     * \returns `small_string<capacity_ + capacity2>` The new small_string with size `capacity_` + `capacity2`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of the strings.
+     */
+    template <size_t capacity2>
+    constexpr small_string<capacity_ + capacity2> operator+(small_string<capacity2> const & rhs) const noexcept
+    {
+        small_string<capacity_ + capacity2> tmp{*this};
+        tmp.insert(tmp.end(), rhs.begin(), rhs.end());
+        return tmp;
+    }
+    //!\}
+};
+
+/*!\name Deduction guides
+ * \{
+ */
+//!\brief Deduces small_string from string literals.
+//!\relates small_string
+template <size_t N>
+small_string(const char (&)[N]) -> small_string<N - 1>;
```

Comment:
```suggestion
small_string(char const (&)[N]) -> small_string<N - 1>;
```

You really have difficulties with east const recently :thinking:

Reply:
that was copy and pasted from renes code (as a tiny defence).